core/src/main/java/org/apache/druid/math/expr/ExprEval.java

```diff
@@ -1129,6 +1032,10 @@ public String asString()
     public boolean isNumericNull()
     {
       if (isScalar()) {
+        if (arrayType.getElementType().is(ExprType.STRING)) {
+          Number n = computeNumber((String) getScalarValue());
```

Comment:
ah yeah, I guess i lost the caching the separate string array impl had. it is probably worth considering adding back, though many of the expression selectors cache their result based on the current offset so that they don't need to recompute the expression when getting the same row

Reply:
This seems a bit different than the use case you described though. The caller of this method will likely call another method to get the actual value after the null check using this method. It would be nice if the getValue method (like `asLong` or `asDouble`) can just return the cached value computed in this method instead of recomputing it.